homeassistant/components/folder_watcher.py

```diff
@@ -0,0 +1,118 @@
+"""
+Component for monitoring activity on a folder.
+
+For more details about this platform, refer to the documentation at
+https://home-assistant.io/components/folder_watcher/
+"""
+import os
+import logging
+import voluptuous as vol
+from homeassistant.helpers.entity import Entity
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['watchdog==0.8.3']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FOLDER = 'folder'
+CONF_PATTERNS = 'patterns'
+CONF_WATCHERS = 'watchers'
+DEFAULT_PATTERN = '*'
+DOMAIN = "folder_watcher"
+EVENT_TYPE = "event_type"
+FILE = 'file'
+FOLDER = 'folder'
+
+WATCHER_CONFIG_SCHEMA = vol.Schema([{
+    vol.Required(CONF_FOLDER): cv.isdir,
+    vol.Optional(CONF_PATTERNS, default=[DEFAULT_PATTERN]): vol.All(
+        cv.ensure_list, [cv.string]),
+}])
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_WATCHERS): WATCHER_CONFIG_SCHEMA,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the folder watcher."""
+    conf = config[DOMAIN]
+    watchers = conf[CONF_WATCHERS]
+
+    def run_setup(event):
+        """"Wait for HA start then setup."""
+        for watcher in watchers:
+            path = watcher[CONF_FOLDER]
+            patterns = watcher[CONF_PATTERNS]
+            if not hass.config.is_allowed_path(path):
+                _LOGGER.error("folder %s is not valid or allowed", path)
+                continue
+            else:
+                Watcher(path, patterns, hass)
```

Comment:
Unnecessary `else` since you `continue` above.

```python
if not hass.config.is_allowed_path(path):
    _LOGGER.error("folder %s is not valid or allowed", path)
    continue
Watcher(path, patterns, hass)
```

Reply:
Hmm the component doesn't setup at all if I do that, will drop the `continue`

Reply:
ok